src/tracer.c

```diff
@@ -40,21 +41,56 @@ tracee_main(char **argv)
 	FATAL_ERROR; // exec shouldn't return.
 }
 
-int
-is_open_syscall(int syscall)
+/*
+    addr is an address pointing to the tracee process' address space, thus we need to copy it.
+*/
+char *
+read_path(char *addr, pid_t pid)
+{
+    long *ptr = (long *)malloc(10000 * sizeof(char));
```

Comment:
since this is path, use MAXPATHLEN

Reply:
Actually... why `malloc()` here and `realloc()` at the end? Since you are realloc-ing, this could be a static buffer.

Reply:
@zvr excuse the use of a magic number, i will fix that and also add bound checking.

Do you mean having a function-internal static buffer of size MAXPATHLEN and then having the caller copy the string instead? Less system calls is always better indeed. I will change it.